controllers/clustertopology_controller_compute_test.go

```diff
@@ -250,6 +252,336 @@ func TestGetClass(t *testing.T) {
 	}
 }
 
+func (f *fakeMachineDeployment) withBootstrapRef(ref *corev1.ObjectReference) *fakeMachineDeployment {
+	f.bootstrapTemplate = ref
+	return f
+}
+func (f *fakeMachineDeployment) withInfraTemplateRef(ref *corev1.ObjectReference) *fakeMachineDeployment {
+	f.infrastructureTemplate = ref
+	return f
+}
+
+func (f *fakeMachineDeployment) withLabels(labels map[string]string) *fakeMachineDeployment {
+	f.labels = labels
+	return f
+}
+func newFakeMachineDeployment(namespace string, name string) *fakeMachineDeployment {
+	return &fakeMachineDeployment{
+		name:      name,
+		namespace: namespace,
+	}
+}
+
+func (f *fakeMachineDeployment) Obj() *clusterv1.MachineDeployment {
+	obj := &clusterv1.MachineDeployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "MachineDeployment",
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      f.name,
+			Namespace: f.namespace,
+			Labels:    f.labels,
+		},
+		Spec: clusterv1.MachineDeploymentSpec{
+			Template: clusterv1.MachineTemplateSpec{
+				Spec: clusterv1.MachineSpec{
+					Bootstrap: clusterv1.Bootstrap{},
+				},
+			},
+		},
+	}
+	if f.bootstrapTemplate != nil {
+		obj.Spec.Template.Spec.Bootstrap.ConfigRef = f.bootstrapTemplate
+	}
+	if f.infrastructureTemplate != nil {
+		obj.Spec.Template.Spec.InfrastructureRef = *f.infrastructureTemplate
+	}
+	return obj
+}
+
+type fakeMachineDeployment struct {
```

Comment:
Let's move the fake object builder definition/implementation down in the file, where all the other builders are defined